pandas/io/common.py

```diff
@@ -1101,6 +1131,23 @@ def _is_binary_mode(handle: FilePath | BaseBuffer, mode: str) -> bool:
     if issubclass(type(handle), text_classes):
         return False
 
-    # classes that expect bytes
-    binary_classes = (BufferedIOBase, RawIOBase)
-    return isinstance(handle, binary_classes) or "b" in getattr(handle, "mode", mode)
+    return isinstance(handle, _get_binary_io_classes()) or "b" in getattr(
+        handle, "mode", mode
+    )
+
+
+def _get_binary_io_classes() -> tuple[type, ...]:
```

Comment:
can you add `.cache` here: https://docs.python.org/3/library/functools.html here

Reply:
It's Python 3.9+ only but I've implemented a simple memoization pattern in the function, WDYT?

Reply:
See discussion below

Reply:
yep this is fine, my concern was that this would hit on *each* file i/o operation when this should just be a static lookup (i know you are fixing this is zstandard but this is what it is now)